tests/math_utils.test.ts

```diff
@@ -85,3 +90,128 @@ describe("remap", () => {
     expect(remap(2, 0, 1, 0, 10, false)).to.equal(20);
   });
 });
+
+describe("getFrameSizeInScreenPx", () => {
+  it("returns canvas size when frame is the same aspect ratio", () => {
+    const frameSizePx = getFrameSizeInScreenPx([100, 100], [1, 1], 1);
+    expect(frameSizePx).to.deep.equal([100, 100]);
+  });
+
+  it("does not change returned size if frame resolution is higher", () => {
+    const frameResolutions: [number, number][] = [
+      [0.1, 0.1],
+      [1, 1],
+      [10, 10],
+      [100, 100],
+    ];
+    for (const frameResolution of frameResolutions) {
+      const frameSizePx = getFrameSizeInScreenPx([100, 100], frameResolution, 1);
+      expect(frameSizePx).to.deep.equal([100, 100]);
+    }
+  });
+
+  it("constrains by height when frame aspect ratio is taller than canvas", () => {
+    const canvasSizePx: [number, number] = [100, 100];
+    const frameResolution: [number, number] = [0.5, 1];
+    const frameZoom = 1;
```

Comment:
nit: but for consistency I would have DEFAULT_ZOOM and DEFAULT_RESOLUTION in these tests anytime you're using 1 so it's easier to read what is being tested and what's a variable. (above you just send `1` into the function without defining it, but here you make it a constant